src/jit/importer.cpp

```diff
@@ -6057,6 +6057,11 @@ GenTreePtr Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolve
                 // In future, it may be better to just create the right tree here instead of folding it later.
                 op1 = gtNewFieldRef(lclTyp, pResolvedToken->hField);
 
+                if (pFieldInfo->fieldFlags & CORINFO_FLG_FIELD_INITCLASS)
+                {
+                    op1->gtFlags |= GTF_FLD_INITCLASS;
```

Comment:
While this location may suffice for the case that we have here,
 I think that it may fail to set the flag on other kinds of GT_FIELD nodes. 
(Like instance fields , can they have or do they need it set?)
(Also what should happen for address of static fields?)